#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_newplugin.dpatch by Thomas Schmidt <tschmidt@debian.org>
##
## Thomas Günther <tom@toms-cafe.de>:
##   - adapted to VDR-1.5.7
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch for the newplugin-script, to not require a normal vdr-source-dir

@DPATCH@
--- vdr-1.5.7/newplugin
+++ vdr-1.5.7/newplugin
@@ -24,7 +24,7 @@
 $PLUGIN_DESCRIPTION = "Enter description for '$PLUGIN_NAME' plugin";
 $PLUGIN_MAINENTRY = $PLUGIN_CLASS;
 
-$PLUGINS_SRC = "PLUGINS/src";
+$PLUGINS_SRC = "./";
 
 $README = qq
 {This is a "plugin" for the Video Disk Recorder (VDR).
@@ -159,7 +159,7 @@
 	\@-rm -rf \$(TMPDIR)/\$(ARCHIVE)
 	\@mkdir \$(TMPDIR)/\$(ARCHIVE)
 	\@cp -a * \$(TMPDIR)/\$(ARCHIVE)
-	\@tar czf \$(PACKAGE).tgz -C \$(TMPDIR) \$(ARCHIVE)
+	\@tar czf \$(PACKAGE).tgz -C \$(TMPDIR) --exclude debian --exclude CVS --exclude .svn \$(ARCHIVE)
 	\@-rm -rf \$(TMPDIR)/\$(ARCHIVE)
 	\@echo Distribution package created as \$(PACKAGE).tgz
 
@@ -311,7 +311,7 @@
 VDRPLUGINCREATOR(cPlugin$PLUGIN_CLASS); // Don't touch this!
 };
 
-$PLUGINDIR = "$PLUGINS_SRC/$PLUGIN_NAME";
+$PLUGINDIR = "$PLUGINS_SRC/$PLUGIN_NAME-$PLUGIN_VERSION";
 
 die "The directory $PLUGINS_SRC doesn't exist!\n" unless (-d "$PLUGINS_SRC");
 die "A plugin named '$PLUGIN_NAME' already exists in $PLUGINS_SRC!\n" if (-e "$PLUGINDIR");
@@ -332,7 +332,8 @@
 * fill in the code skeleton in "$PLUGIN_NAME.c" to implement your plugin function
 * add further source files if necessary
 * adapt the "Makefile" if necessary
-* do "make plugins" from the VDR source directory to build your plugin
+* do "make all DVBDIR=/usr VDRDIR=/usr/include/vdr LIBDIR=." from the source
+  directory to build your plugin
 
 };
 
